src/bid.rs

```diff
@@ -278,15 +278,20 @@ impl Bid {
         self.expiration
     }
 
+    /// Returns a mutable ref pointing to the `pos` field of the
+    /// Bid.
+    pub fn expiration_mut(&mut self) -> &mut u64 {
```

Comment:
Why is `expiration_mut` implemented differently than `set_pos`?

Reply:
Because we never need to set it, we always need to do `+=` ops on it. Therefore, `set_position` is not needed.

Reply:
In that case, the function should be `pub fn extend_expiration(&mut self, extend: u64) { self.expiration += extend }`

Reply:
What if I want in the tests to reduce it? I think the other is more flexible TBH..

Reply:
I don't think that adding a functionality just to be able to break the whole structure is a good trade-off.

If the purpose is testing then we should expose the attribute to the crate (e.g. `pub(crate)`) instead of making public functions that will not behave as expected

Reply:
The tests of the contracts are not performed in this lib. But require to mutate on this way too.

Reply:
And again, I don't see why this is wrong or needs to require that much discussion. In the sense of: Follows the API guidelines and I don't think there's the need of extremely constraining the function. At the end it's gonna be used only by us in the Genesis contracts..

Reply:
I'm inclined to agree with @CPerezz, I don't see the point in having an extensive discussion about such a minute detail - furthermore, if this keeps us flexible for now (both in ability to test and less chance of API breakage) then I think, in this stage, it's better to keep it as is.